### PR TITLE
Schema generation for ADI2.

### PR DESCRIPTION
**Pull request by AndyPerryKainos**
# Description and relevant Jira numbers

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: mes-test-schema/categories/ADI2/index.json [hunk 1]
```diff
@@ -0,0 +1,1822 @@
+{
+	"title": "Test Result ADI Part 2 Schema",
+	"type": "object",
+	"definitions": {
+		"code78Present": {
```
Comment (appleabs) [inline mes-test-schema/categories/ADI2/index.json]: i don't think adi has code 78

Comment (AndyPerryKainos) [reply mes-test-schema/categories/ADI2/index.json]: Is that index.json even relevant? code78 doesn't appear in the generated type definition nor is it included in the source partial. From what I can see the index.json file is just an aggregation of stuff from common and the partial. 

E.g. code78 exists in the json file for other categories where it is not applicable. 

Happy for someone to explain this if it needs changing.

Comment (appleabs) [reply mes-test-schema/categories/ADI2/index.json]: ah ok didn't realise code 78 was in common

File: mes-test-schema/categories/ADI2/partial.d.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+  };
+}
+/**
+ * Details of the Tell Me questions asked during the test
+ *
+ * This interface was referenced by `PartialTestResultADI2Schema`'s JSON-Schema
+ * via the `definition` "vehicleChecks".
+ */
+export interface VehicleChecks {
+  tellMeQuestions?: {
```
Comment (Cavalier07) [inline mes-test-schema/categories/ADI2/partial.d.ts]: i think we should stick with arrays on this. The reason it's not on Cat B is a tech debt item we want to look at but can't as it's a breaking change

Comment (AndyPerryKainos) [reply mes-test-schema/categories/ADI2/partial.d.ts]: I was careful to make sure it is an array.

File: mes-test-schema/categories/ADI2/partial.d.ts [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    code?: string;
+    /**
+     * Description of the question that was asked
+     */
+    description?: string;
+    /**
+     * Outcome of the question that was asked
+     */
+    outcome?: "P" | "DF" | "S" | "D";
+  }[];
```
Comment (AndyPerryKainos) [inline mes-test-schema/categories/ADI2/partial.d.ts]: It is an array, but it looks like I need to tweak it to use the QuestionResult rather that this literal.

Comment (AndyPerryKainos) [reply mes-test-schema/categories/ADI2/partial.d.ts]: Fixed.

File: mes-test-schema/categories/ADI2/partial.d.ts (hunk 2 above)
Comment (AndyPerryKainos) [reply mes-test-schema/categories/ADI2/partial.d.ts]: Updated to use array of QuestionResult.

**Pull request merged**